hil/ext/switches/dell.py

```diff
@@ -55,6 +57,18 @@ def validate(kwargs):
     def session(self):
         return _PowerConnect55xxSession.connect(self)
 
+    @staticmethod
+    def validate_port_name(port):
+        """
+        Valid port names for this switch are of the form gi1/0/11,
+        te1/0/12, gi1/12, or te1/3
+        """
+
+        val = re.compile('(^(gi|te)\d+[/]\d+[/]\d+$)|(^(gi|te)\d+[/]\d+$)')
```

Comment:
Two comments about the regexes (not just this one):

* Use [raw strings][1], so we don't have to worry about multiple levels of escaping.
* I would just use the optional `?` operator, rather than repeating the whole pattern for 2 s 3 numbers.
* I'm pretty sure you don't need to escape slashes.

So for this one, I would have:

```python
val  = re.compile(r'^(gi|te)\d+/\+d(/\d+)?$')
```

[1]: https://docs.python.org/2/reference/lexical_analysis.html#string-literals